modules/xmpp/JingleSessionPC.js

```diff
@@ -501,205 +302,46 @@ JingleSessionPC.prototype.setRemoteDescription = function (elem, desctype) {
-        for (i = 0; i < self.remoteSDP.media.length; i++) {
-            if (SDPUtil.find_line(self.remoteSDP.media[i], 'a=mid:' + $(this).attr('name')) ||
-                self.remoteSDP.media[i].indexOf('m=' + $(this).attr('name')) === 0) {
-                idx = i;
-                break;
-            }
-        }
-        if (idx == -1) { // fall back to localdescription
-            for (i = 0; i < self.localSDP.media.length; i++) {
-                if (SDPUtil.find_line(self.localSDP.media[i], 'a=mid:' + $(this).attr('name')) ||
-                    self.localSDP.media[i].indexOf('m=' + $(this).attr('name')) === 0) {
-                    idx = i;
-                    break;
-                }
-            }
-        }
-        var name = $(this).attr('name');
-        // TODO: check ice-pwd and ice-ufrag?
-        $(this).find('transport>candidate').each(function () {
-            var line, candidate;
-            var protocol = this.getAttribute('protocol');
-            protocol =
-                (typeof protocol === 'string') ? protocol.toLowerCase() : '';
-            if ((self.webrtcIceTcpDisable && protocol == 'tcp') ||
-                (self.webrtcIceUdpDisable && protocol == 'udp')) {
-                return;
-            }
-
-            line = SDPUtil.candidateFromJingle(this);
-            candidate = new RTCIceCandidate({sdpMLineIndex: idx,
-                sdpMid: name,
-                candidate: line});
-            try {
-                self.peerconnection.addIceCandidate(candidate);
-            } catch (e) {
-                logger.error('addIceCandidate failed', e.toString(), line);
-                self.room.eventEmitter.emit(XMPPEvents.ADD_ICE_CANDIDATE_FAILED,
-                    err, self.peerconnection);
-            }
-        });
-    });
-};
-
-JingleSessionPC.prototype.sendAnswer = function (provisional) {
-    //logger.log('createAnswer', provisional);
-    var self = this;
+JingleSessionPC.prototype.sendAnswer = function (success, failure) {
+    //logger.log('createAnswer');
     this.peerconnection.createAnswer(
         function (sdp) {
-            self.createdAnswer(sdp, provisional);
-        },
-        function (e) {
-            logger.error('createAnswer failed', e);
-            self.room.eventEmitter.emit(XMPPEvents.CONFERENCE_SETUP_FAILED);
-        },
+            this.createdAnswer(sdp, success, failure);
+        }.bind(this),
+        function (error) {
+            logger.error("createAnswer failed", error);
+            if (failure)
+                failure(error);
+            this.room.eventEmitter.emit(
+                    XMPPEvents.CONFERENCE_SETUP_FAILED, error);
+        }.bind(this),
         this.media_constraints
     );
 };
 
-JingleSessionPC.prototype.createdAnswer = function (sdp, provisional) {
+JingleSessionPC.prototype.createdAnswer = function (sdp, success, failure) {
     //logger.log('createAnswer callback');
-    var self = this;
+    const self = this;
```

Comment:
I'm not sure if "const" should work on all the browsers but maybe it's better to change it to "var" for code consistency.